fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -153,12 +189,37 @@ public InputStream getBinaryContent(final String identifier, final Instant versi
     }
 
     @Override
-    public void commit() throws PersistentStorageException {
+    public void commit(final CommitOption option) throws PersistentStorageException {
         if (isReadOnly()) {
             // No changes to commit.
             return;
         }
-        // commit changes.
+
+        try {
+            mutex.lock();
+            //prepare session for commit
+            final Collection<OCFLObjectSession> sessions = this.sessionMap.values();
+
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.prepare();
+            }
+
+            //perform commit
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.commit(option);
+            }
+
+            //close each session
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.close();
+            }
+
+            //purge object sessions
+            sessionMap.clear();
+
+        } finally {
+            mutex.unlock();
```

Comment:
The rollback scenario  for failing during prepare() as well as during object commits is fairly straightforward when committing new versions.  When committing to the mutable head,  I'm not sure how we rollback if a commit fails.

Reply:
I don't think you have to address rollback in this PR, since its likely to be involved even in the "new version" scenario. I just wasn't sure where it was going to be trigger. Like would this method call the rollback itself on failure, or would whatever client was calling the commit catch the exception and trigger the rollback.

Reply:
I'm going to take a swing at a partial rollback implementation.  I'm thinking if the commit fails, an exception will be thrown.  Then the user will call rollback().   Here's a JIRA for completing the rollback implementation: https://jira.duraspace.org/browse/FCREPO-3130

Reply:
So looks like the answer is that whatever called `commit` will need to call `rollback`. I am guessing the flow will be that if any exception is thrown during commit, rollback should be called? Just checking to make sure the `State` of the session doesn't need to be exposed.

Reply:
That's how I'm thinking about it.  No need to expose State.